src/kademlia/node.cpp

```diff
@@ -115,9 +98,16 @@ node::node(udp_socket_interface* sock
 	, m_last_self_refresh(min_time())
 	, m_sock(sock)
 	, m_counters(cnt)
+	, m_storage(dht_default_storage_constructor(m_id, m_settings, m_counters))
 {
 	m_secret[0] = random();
 	m_secret[1] = random();
+
+	TORRENT_ASSERT(m_storage != NULL);
+}
+
+node::~node() {
+	if (m_storage != NULL) delete m_storage;
```

Comment:
I would much rather see m_storage be a boost::scoped_ptr and not manually destruct it in the destructor. (btw, delete NULL; is fine, there's no need to check for it).